sdk/metricsadvisor/ai-metrics-advisor-rest/src/metricsAdvisorKeyCredentialPolicy.ts

```diff
@@ -0,0 +1,106 @@
+  PipelineResponse,
+  SendRequest,
+} from "@azure/core-rest-pipeline";
+import { KeyCredential } from '@azure/core-auth';
+export const API_KEY_HEADER_NAME = "Ocp-Apim-Subscription-Key";
+export const X_API_KEY_HEADER_NAME = "x-api-key";
+
+/**
+ * Interface parameters for updateKey function
+ */
+export interface MetricsAdvisorKeys extends KeyCredential {
+  /** API key from the Metrics Advisor web portal */
+  // key?: string;
+  /** Subscription access key from the Azure portal */
+  subscriptionKey?: string;
+}
+
+/**
+ * Credential used to authenticate and authorize with Metrics Advisor service
+ */
+export class MetricsAdvisorKeyCredential implements MetricsAdvisorKeys {
+  /**
+   * Subscription access key from the Azure portal
+   */
+  private _subscriptionKey: string;
+
+  /**
+   * API key from the Metrics Advisor web portal
+   */
+  private _apiKey: string;
+
+  /**
+   * Creates an instance of MetricsAdvisorKeyCredential
+   *
+   * @param subscriptionKey - Subscription access key from the Azure portal
+   * @param apiKey - API key from the Metrics Advisor web portal
+   */
+  constructor(subscriptionKey: string, apiKey: string) {
+    if (!subscriptionKey || !apiKey) {
+      throw new RangeError("Subscription Key or Api Key doesn't have a valid value");
+    }
+    this._subscriptionKey = subscriptionKey;
+    this._apiKey = apiKey;
+  }
+
+  /**
+   * Get Api Key
+   */
+  public get key(): string {
+    return this._apiKey;
+  }
+
+  /**
+   * Get Subscription key
+   */
+  public get subscriptionKey(): string {
+    return this._subscriptionKey;
+  }
+
+  /**
+   * Change the value of the subscription and/or api keys.
+   *
+   * Updates will take effect upon the next request after
+   * updating the key value.
+   *
+   * @param subscriptionKey - The new subscription key value to be used
+   * @param apiKey - The new api key value to be used
+   */
+  public updateKey(metricAdvisorKeys: MetricsAdvisorKeys): void {
+    if (!metricAdvisorKeys.subscriptionKey && !metricAdvisorKeys.key) {
+      throw new Error("At least one of the subscriptionKey and apiKey must be a non-empty string");
+    }
+    if (metricAdvisorKeys.subscriptionKey) {
+      this._subscriptionKey = metricAdvisorKeys.subscriptionKey;
+    }
+    if (metricAdvisorKeys.key) {
+      this._apiKey = metricAdvisorKeys.key;
+    }
+  }
+}
+
+/**
+ * Creates an HTTP pipeline policy to authenticate a request
+ * using an `MetricsAdvisorKeyCredential`
+ */
+export function createMetricsAdvisorKeyCredentialPolicy(
+  credential: MetricsAdvisorKeyCredential
+): PipelinePolicy {
+  return {
+    name: "metricsAdvisorKeyCredentialPolicy",
+    sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      if (!request) {
+        throw new Error("webResource cannot be null or undefined");
+      }
+      request.headers.set(API_KEY_HEADER_NAME, credential.subscriptionKey);
+      request.headers.set(X_API_KEY_HEADER_NAME, credential.key);
```

Comment:
@joheredi Add a policy to add additional policy

Reply:
@azure-rest/core-client can already handle key credentials. However, the current implementation only sets one header. I think we can update core-client-rest to handle `Ocp-Apim-Subscription-Key` as it seems to be common enough. I'm preparing a PR for this. We should be able to remove this class altogether.

Reply:
We are already capable of handling Key Credentials in RLC without needing custom code. However the handling we have only supports a single key header. I wonder if this is a common pattern for key authentication and if it is worth extending the KeyCredential.

Let me spend some time thinking about this and I'll come back to you on this. Meanwhile keep this policy.

One quick comment though, we are trying to not use classes as much as possible in RLC. Please take a look at this policy I think the implementation would be very similar:

https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-client-rest/src/keyCredentialAuthenticationPolicy.ts